debian/ubuntu-advantage-tools.postinst

```diff
@@ -302,6 +304,13 @@ notify_wrong_fips_metapackage_on_cloud() {
     fi
 }
 
+remove_old_systemd_timers() {
+    if systemctl list-timers --all | grep -q "$OLD_MESSAGING_TIMER"; then
+        rm -f $OLD_MESSAGING_TIMER_MASKED_LOCATION
+        systemctl daemon-reload
+    fi
+}
```

Comment:
Could we just test for the presence of the OLD_MESSAGING_TIMER_MASKED_LOCATION? Also most post(inst|rm) scripts I see also silence the logs from the reload (in case other systemd units on the system  are malformed, we don't want to generate a lot of logs to the console during our pkg upgrade. We also don't want to be impacted during our package install of a failure to run systemctl daemon-reload. So let's try something like this 
`systemctl --system daemon-reload >/dev/null || true`



```suggestion
    if [ -f  $OLD_MESSAGING_TIMER_MASKED_LOCATION ]; then
        rm $OLD_MESSAGING_TIMER_MASKED_LOCATION
        systemctl --system daemon-reload >/dev/null || true
    fi
}
```

Reply:
That does appear to be a popular pattern +1 https://codesearch.debian.net/search?q=systemctl+--system+daemon-reload+%3E%2Fdev%2Fnull+%7C%7C+true+path%3Adebian%2F&literal=1&page=1

Reply:
Actually I think debhelper has what we need, its in the postrm of the last version of uaclient:
```
# Automatically added by dh_systemd_enable
if [ "$1" = "remove" ]; then
        if [ -x "/usr/bin/deb-systemd-helper" ]; then
                deb-systemd-helper mask ua-messaging.timer >/dev/null
        fi
fi

if [ "$1" = "purge" ]; then
        if [ -x "/usr/bin/deb-systemd-helper" ]; then
                deb-systemd-helper purge ua-messaging.timer >/dev/null
                deb-systemd-helper unmask ua-messaging.timer >/dev/null
        fi
fi
# End automatically added section
```

I think we need to run the "purge" section here.

Reply:
Indeed we have that, the problem is that we will only have that part of the code running if the user manually runs an `apt purge ubuntu-advantage-tools`

I don't think this is called during the upgrade process of the package

Reply:
Also fun fact: `systemctl --system daemon-reload >/dev/null || true` is already added to our postinst by debhelper, so we shouldn't have to call it ever explicitly

Reply:
This looks really good. Thank you for fleshing this out and coming up with a better solution for this rename. +1, if we think we want to got ahead and "fix" this issue of stale/masked timers across upgrade. Doesn't break downgrade compatibility as older versions will re-create the timer/service.